### PR TITLE
US091 - Find respondent by email

### PR DESCRIPTION
**Pull request by gemmai95**
Changed the endpoint to work with email in json rather than URL and to return only verified users this works with ROps and backstage.

### CODE REVIEW

File: test/test_data/mock_respondent.py [hunk 1]
```diff
@@ -77,3 +77,30 @@ def as_party(self):
...
+        self._attributes.update(kwargs)
+        return self
+
+    def as_respondent(self):
+        return self._attributes
+
+    def as_party(self):
+        props, attrs = partition_dict(self._attributes, ['id', 'sampleUnitType'])
+        props['attributes'] = attrs
+        return props
```
Comment (jcox-dev) [inline test/test_data/mock_respondent.py]: Missing newline.

File: .pytest_cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (jcox-dev) [inline .pytest_cache/v/cache/lastfailed]: This file can probably be removed.

File: ras_party/controllers/queries.py [hunk 3]
```diff
@@ -52,6 +52,18 @@ def query_respondent_by_email(email, session):
     return session.query(Respondent).filter(func.lower(Respondent.email_address) == email.lower()).first()
 
 
+def query_respondent_by_email_filter_out_created(email, session):
+    """
+    Query to return respondent based on email
+    :param email: the party uuid
+    :return: respondent or none
+    """
+    logger.debug('Querying respondents by email', email=email)
```
Comment (jcox-dev) [inline ras_party/controllers/queries.py]: Are we ok to log out email addresses?

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Looks good with just a couple of tiny considerations 👍

File: .pytest_cache/v/cache/lastfailed (hunk 2 above)
Comment (jwgwalton) [reply .pytest_cache/v/cache/lastfailed]: Awwww, I liked this file.

File: test/party_client.py [hunk 4]
```diff
@@ -140,8 +140,11 @@ def get_respondent_by_id(self, id, expected_status=200):
         self.assertStatus(response, expected_status, "Response body is : " + response.get_data(as_text=True))
         return json.loads(response.get_data(as_text=True))
 
-    def get_respondent_by_email(self, email, expected_status=200):
-        response = self.client.get(f'/party-api/v1/respondents/email/{email}', headers=self.auth_headers)
+    def get_respondent_by_email(self, payload, expected_status=200):
+        response = self.client.get('/party-api/v1/respondents/email',
+                                   data=json.dumps(payload),
+                                   headers=self.auth_headers,
+                                   content_type='application/vnd.ons.business+json')
```
Comment (jwgwalton) [inline test/party_client.py]: Can it not just be application/json?

File: ras_party/controllers/queries.py [hunk 5]
```diff
@@ -52,6 +52,18 @@ def query_respondent_by_email(email, session):
...
+
+    return session.query(Respondent).filter(and_(func.lower(Respondent.email_address) == email.lower(),
+                                                 Respondent.status != 'CREATED')).first()
```
Comment (jwgwalton) [inline ras_party/controllers/queries.py]: There are 3 possible statuses: CREATED, ACTIVE, SUSPENDED. Did we only want active respondents?

File: ras_party/controllers/queries.py (hunk 3 above)
Comment (gemmai95) [reply ras_party/controllers/queries.py]: Damn thought I caught all them. Good catch

File: ras_party/controllers/queries.py (hunk 5 above)
Comment (gemmai95) [reply ras_party/controllers/queries.py]: We just don't want created ones.

**Comment by gemmai95** [comment]
[US091: Internal user to search respondent via their email address](https://trello.com/c/fpJMbZOk/62-us091-internal-user-to-search-respondent-via-their-email-address)

**Review by jcox-dev — APPROVED** [review]
Other than the [missing newline](https://github.com/ONSdigital/ras-party/pull/120/files/7ddc1633d8a984b00af87ae3cee4a93aceb8f25d#r177444581) the tests all pass and it LGTM 👍

**Review by jwgwalton — APPROVED** [review]

**Review by rhiGriff — APPROVED** [review]

File: ras_party/controllers/queries.py [hunk 6]
```diff
@@ -47,11 +47,23 @@ def query_respondent_by_email(email, session):
     :param email: the party uuid
     :return: respondent or none
     """
-    logger.debug('Querying respondents by email', email=email)
+    logger.debug('Querying respondents by email')
```
Comment (rhiGriff) [inline ras_party/controllers/queries.py:50]: 👍

**Pull request merged**